theano/scan_module/tests/test_scan.py

```diff
@@ -625,12 +625,11 @@ def f_rnn(u_t, x_tm1, W_in, W):
         utt.assert_allclose(theano_out1, v_out1)
         utt.assert_allclose(theano_out2, v_out2)
 
-        topo = f2.maker.fgraph.toposort()
-        scan_node = [node for node in topo
-                     if isinstance(node.op, theano.scan_module.scan_op.Scan)]
-        assert len(scan_node) == 1
-        scan_node = scan_node[0]
-        assert scan_node.op.gpu
+        # Ensure f2 contains only one scan node and it's been moved to the GPU
+        scan_nodes = self.scan_nodes_from_fct(f2)
+        assert len(scan_nodes) == 1
+        assert (scan_nodes[0].op.info.get('gpu', False) or
+                scan_nodes[0].op.info.get('gpua', False))
```

Comment:
I would check for gpu only here since the function uses mode_with_gpu.  This shouldn't get moved to gpua.